metaflow/plugins/argo/argo_workflows.py

```diff
@@ -114,19 +132,42 @@ def __init__(
         self.max_workers = max_workers
         self.workflow_timeout = workflow_timeout
         self.workflow_priority = workflow_priority
+        self._ignore_events = ignore_events
+        self._triggering_flows = []
+        self._triggering_events = []
+        self._reset_at = None
+
+        # The values with curly braces '{{}}' are made available by Argo
+        # Workflows. Unfortunately, there are a few bugs in Argo which prevent
+        # us from accessing these values as liberally as we would like to - e.g,
+        # within inline templates - so we are forced to generate container templates
+        self._run_id = "argo-{{workflow.name}}"
 
         self.parameters = self._process_parameters()
+        self._sensor_template = self._compile_sensor_template()
         self._workflow_template = self._compile()
         self._cron = self._cron()
 
     def __str__(self):
         return str(self._workflow_template)
 
+    def sensor_template(self):
+        return self._sensor_template
+
     def deploy(self):
         try:
-            ArgoClient(namespace=KUBERNETES_NAMESPACE).register_workflow_template(
+            client = ArgoClient(namespace=KUBERNETES_NAMESPACE)
+            client.register_workflow_template(
                 self.name, self._workflow_template.to_json()
             )
+            if are_events_configured():
+                if self._sensor_template is not None:
+                    client.register_sensor_template(
+                        self._sensor_template.name,
+                        self._sensor_template.to_json(),
+                    )
+                else:
+                    client.disable_sensor(format_sensor_name(self.name))
```

Comment:
nit - sensor name could be simply `self.name` and (minor nit)`client.register_sensor_template(name, payload=None)` can perhaps take care of unsetting the sensor if the payload is missing.

Reply:
Cleaned this up a bit. We can't use `self.name` for the sensor name when NATS or Jetstream is enabled. Argo Events uses the sensor name to establish a connection which also seems to allocate some per-client storage in the messaging cluster (buffer perhaps?). In any case, the messaging cluster complains when the sensor name contains `.`. 

As a workaround I changed the sensor naming code to just replace `.` with `-` and dropped the `mf` and `trigger` prefix and suffix.